tests/unit/Endpoints/Webhooks/ProductHandlerTest.php

```diff
@@ -68,58 +47,121 @@ public function testCreateProduct()
+    {
+        $woocommerceUpdatedProduct = $this->mockUpdateProductRequestWithTest();
+
+        // Get the updated data dump as a dotnotated collection
+        $updatedFile = $this->getDataDump(self::UPDATE_DATADUMP_DIRECTORY.'/'.self::UPDATE_DATADUMP_PRODUCT);
+        $updatedProductData = $updatedFile->getReturn()['data'];
+        $updatedProductData = new Dot($updatedProductData[0]);
+
+        // Compare the values of the updated data dump against the updated wordpress product
+        $this->assertProduct($updatedProductData, $woocommerceUpdatedProduct, false);
+        $this->assertTaskNotCount(0, 'There are suppose to be any tasks');
+
+        $sku = $woocommerceUpdatedProduct->get_sku();
+        // Stock quantity should not be updated
+        $expectedStockQuantity = $updatedProductData->get('flat_product.product.product_stock.value');
+        $this->assertNotEquals(
+            $expectedStockQuantity,
+            $woocommerceUpdatedProduct->get_stock_quantity(),
+            "[sku=$sku] WooCommerce stock quantity should not match"
+        );
+
+        // Regular price should not be updated
+        if (self::WC_TYPE_CONFIGURABLE !== $woocommerceUpdatedProduct->get_type()) {
+            $expectedRegularPrice = $updatedProductData->get('product_default_price.ppu_wt');
+            $this->assertNotEquals(
+                $expectedRegularPrice,
+                $woocommerceUpdatedProduct->get_regular_price(),
+                "[sku=$sku] WooCommerce regular price should not match expected regular price"
+            );
+
+            // Discounted price should not be updated
+            if ($updatedProductData->get('product_price.ppu_wt') !== $expectedRegularPrice) {
+                $expected_discounted_price = $updatedProductData->get('product_price.ppu_wt');
+                $this->assertNotEquals(
+                    $expected_discounted_price,
+                    $woocommerceUpdatedProduct->get_sale_price(),
+                    "[sku=$sku] WooCommerce discount price should not match the expected discount price"
+                );
+            }
+        }
+    }
+
+    protected function subtestStockOnly(): void
+    {
+        $woocommerceUpdatedProduct = $this->mockUpdateProductRequestWithTest();
+
+        // Get the updated data dump as a dotnotated collection
+        $updatedFile = $this->getDataDump(self::UPDATE_DATADUMP_DIRECTORY.'/'.self::UPDATE_DATADUMP_PRODUCT);
+        $updatedProductData = $updatedFile->getReturn()['data'];
+        $updatedProductData = new Dot($updatedProductData[0]);
+
+        // Compare the values of the updated data dump against the updated wordpress product
+        $this->assertProduct($updatedProductData, $woocommerceUpdatedProduct, false);
+
+        $sku = $woocommerceUpdatedProduct->get_sku();
+        $this->assertProductStock($updatedProductData, $woocommerceUpdatedProduct, $sku);
+
+        $this->assertTaskNotCount(0, 'There are suppose to be any tasks');
```

Comment:
I don't understand this assert, can you fix the message so it's clear?
Also on the other places where this message is used

Reply:
Changed to `There are not supposed to be any tasks left to process`.